tests/integration/session_comprehension_head/test.rflx

```diff
@@ -0,0 +1,57 @@
+with Universal;
+
+package Test is
+
+   generic
+      Channel : Channel with Writable; -- §S-P-C-W
+   session Session with
+      Initial => Start,
+      Final => Terminated
+   is
+      Options : Universal::Options; -- §S-D-V-T-MS, §S-D-V-E-N
+      First_Option : Universal::Option; -- §S-D-V-T-M, §S-D-V-E-N
+   begin
+      state Start is
+      begin
+         -- §S-S-A-AP-MA, §S-E-A-T-SC, §S-E-A-E-L
+         Options'Append (Universal::Option'(Option_Type => Universal::OT_Data, Length => 1, Data => [2]));
+         -- §S-S-A-AP-MA
+         Options'Append (Universal::Option'(Option_Type => Universal::OT_Null));
+         -- §S-S-A-AP-MA, §S-E-A-T-SC, §S-E-A-E-L
+         Options'Append (Universal::Option'(Option_Type => Universal::OT_Data, Length => 2, Data => [2, 3]));
+      transition
+         goto Process -- §S-S-T-N
+      exception
+         goto Terminated -- §S-S-E
+      end Start;
+
+      state Process is
+         First_Option_Length : Universal::Length; -- §S-S-D-V-T-SC, §S-S-D-V-E-H
+      begin
+         -- Needed for SPARK if the list comprehension does not return an element.
+         -- §S-S-A-A-L
+         First_Option_Length := 0;
+         -- §S-S-A-A-LC, §S-S-A-A-HAT, §S-E-AT-H-SS, §S-E-AT-H-LC, §S-E-S-LC, §S-E-S-BE, §S-E-LC-SSS, §S-E-LC-TSS
+         First_Option_Length := [for E in Options if E.Option_Type = Universal::OT_Data => E.Length]'Head;
```

Comment:
Is there a good reason for not doing the initialization in the generated code? I don't think it's a good language design to require an initialization in such a case. Instead of adding a special case, we could also change the code generation so that local variables are always initialized with `<Variable_Type>'First`. We already do this for global variables. In the long term, the model verification should detect cases where a variable is used before its initialization.

Reply:
I changed the generator so that in case of a scalar type a default value is set when the found flag is set to false. This keeps the correct behaviour for unused assignments and can be done with local information available to the generator.